cmake/linker.cmake

```
@@ -0,0 +1,6 @@
+include_guard(GLOBAL)
+
+find_program(MOLD_BIN mold)
+if(MOLD_BIN AND (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+    add_link_options("-fuse-ld=mold")
```

This will add `mold` to the global linker setting, which may be ok but will also cause this to propagate to subproject within the build (which may or may not be desirable).  However, we depend on a number of libraries via the CMake method `FetchContent`: https://cmake.org/cmake/help/latest/module/FetchContent.html.  `FetchContent` allow overriding package setting so the project generally keeps all settings within targets.

The `morpheus::config` target is the root target of the whole project so it would probably make sense to add it to this target as compiler setting currently: https://github.com/Twon/Morpheus/blob/fa14cd20929517cf2d200065715e7ae25762003f/cmake/compiler.cmake#L6

Note for the `morpheus::config` library to be visible here you may need to add an `include(compiler) in here, but that should be fine as the include guard should avoid that causing any issues.
```suggestion
    target_link_options(morpheus::config
        PUBLIC
             "-fuse-ld=mold"
    )
```

---

Thanks for the explanation!